src/QMCHamiltonians/EwaldTools.h

```diff
@@ -0,0 +1,382 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2019 QMCPACK developers.
+//
+// File developed by: Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
+//
+// File created by: Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef EWALD_TOOLS_H
+#define EWALD_TOOLS_H
+
+#include "QMCHamiltonians/EwaldRef.h"
+
+namespace qmcplusplus
+{
+
+  using namespace ewaldref;
```

Comment:
When adding new code to the source, how do you determine what needs separate namspace(s) and what does not?  Most of the code simply falls within the qmcplusplus namespace (which really should be `qmcpack` since the name of the code changed from QMC++ long ago...).

Reply:
There are two topics.
1. header file "using namespace" is a bad practice. cpp file is fine. "using namespace" propagates in header files but not in cpp files.
2. There is no strict rule whether extra namespace should be used or not. The original EwaldRef.h defines a few datatypes directly in the qmcplusplus name space. They can propagate and can be misused or cause potential name collisions. So I put most of the content of EwaldRef.h in ewaldref namespace within qmcplusplus namespace as a protection. Here the added code is quite standalone and closely related to EwaldRef then putting it in the same namespace as EwaldRef seems appropriate.

Reply:
I'm fine with this, though the new code is not going to be used as reference.  If it has a future, it will be in the main application, otherwise it will be deleted.

Reply:
If you intend to implement a class derived from OperatorBase, then don't put the class in ewaldref namespace. Instead, you can either put `ewaldref::` for using any piece from EwaldRef or "using ewaldref::ABC" in the class to bring in ABC to the new class.